mega.go

```diff
@@ -1617,6 +1617,11 @@ func (m *Mega) UploadFile(srcpath string, parent *Node, name string, progress *c
 		return nil, err
 	}
 
+	err = infile.Close()
```

Comment:
That would be better done in a `defer func() { ... }` just after the `OpenFile` I think

Reply:
but how to return error in defer?

Reply:
Name the return parameters

```go
func (m *Mega) UploadFile(srcpath string, parent *Node, name string, progress *chan int) (node *Node, err error) {
```

Then you can refer to `err` in the `defer func()` something like this which takes care not to overwrite err if it is already set.

```go
defer func() {
    closeErr := infile.Close()
    if closeErr != nil && err == nil {
        err = closeErr
    }
}()
```

Reply:
then error with closing could be lost if there will be error in return statement, but in current case it's important to call u.Finish(), so I refactored my fix, thank you